one-ui/src/main/java/com/marklogic/hub/oneui/auth/LoginLogoutHandler.java

```diff
@@ -43,19 +42,34 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
         ConnectionAuthenticationToken authenticationToken = (ConnectionAuthenticationToken)authentication;
         ObjectNode resp = mapper.createObjectNode();
         boolean isInstalled = authenticationToken.stagingIsAccessible();
+        final boolean[] managePrivilege = new boolean[3];
         if (authenticationToken.stagingIsAccessible()) {
             try {
                 RolesService rolesService = RolesService.on(authenticationToken.getHubConfigSession().newStagingClient());
                 resp.putArray("roles").addAll((ArrayNode) rolesService.getRoles());
+                if (authenticationToken.hasManagePrivileges()) {
+                    ArrayNode rolesNodes = (ArrayNode) resp.get("roles");
```

Comment:
I don't think this logic is really being exercised by the test, right? i.e. there's just a test for an admin and a test for a non-admin.

I recommend instead moving this logic into a protected method and then writing unit tests (i.e. don't extend HubTestBase) that test the logic in here. The protected method would be like this:

    protected boolean userCanInstall(ArrayNode roles)

The scenarios would then be:

1. User has admin
1. User has manage-admin and security, but not admin
1. User has manage-admin but not security nor admin
1. User has security but not manage-admin nor admin
1. User has none of the 3 roles 

And note the method returns a single boolean - that simplifies the code below, where it doesn't need to look at an array of 3 booleans any more.

Reply:
@rjrudin only 1 or 2 or both can install ? right?

Reply:
Story says user with role manage-admin and security should be able to install. 
So for the test 1 and 2 should be able to install.

Reply:
Right, 1 and 2 are true and 3/4/5 are false.